### DISCUSSION TITLE
JSON Library

### DISCUSSION THREAD

**Discussion by mths1** [vernemq/vernemq#2114] [General]
VerneMQ uses jsx as JSON library. According to those benchmarks 
https://github.com/sile/jsone/blob/master/benchmark/README.md

JSX seems to be pretty slow, compared to other choices esp.

https://github.com/davisp/jiffy

(Furthermore, I am not sure if JSX is still maintained. Jiffy and jsone) seem to be.

JSON seems to be nicely encapsulated in VerneMQ, so switching the library shouldn't be a big deal (of course, there might be "details" around UTF-8 etc.)

**Comment by ioolkos** [comment]
Yeah, that's why there's the small `vmq_json` module.
Another performance oriented Erlang was recently announced too, I'll have to find it.
Changeing this should be pretty low effort. We have to ensure things work with the JSON in `vmq_diversity` (Lua/Luerl stuff)

@codeadict not sure but we might have talked about this & maybe you have a recommendation too.

**Comment by ioolkos** [comment]
Ah, yes it was Thoas: https://github.com/lpil/thoas
I haven't looked at it, though. It does not have C sources, that's the drawback of Jiffy. I'd prefer to avoid that.

**Comment by codeadict** [comment]
Yeah i would avoid any NIF based library, thoas and jsone are pretty good

**Comment by codeadict** [comment]
RabbitMQ is using Thoas since the last version and they are contributing to it. Its basically a port of Elixir's Jason library o Erlang
